staging/src/k8s.io/apiserver/pkg/endpoints/handlers/apply.go

```diff
@@ -17,164 +17,80 @@ limitations under the License.
-	if m == nil {
-		m = make(map[string]metav1.VersionedFieldSet)
-	}
-	// TODO: save the managedFields correctly
-	m[workflow] = metav1.VersionedFieldSet{
-		APIVersion: string(j),
+		return nil, err
 	}
-	accessor.SetManagedFields(m)
-	return nil
-}
 
-func (p *applyPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (runtime.Object, error) {
-	current, err := p.convertCurrentVersion(currentObject)
+	currentTyped, err := p.converter.ObjectToTyped(currentObject)
 	if err != nil {
-		return nil, fmt.Errorf("failed to convert current object: %v", err)
+		return nil, fmt.Errorf("failed to create typed current object: %v", err)
 	}
 
-	lastIntent, err := p.extractLastIntent(currentObject, workflowId)
+	newTyped, err := p.converter.ObjectToTyped(newObject)
 	if err != nil {
-		return nil, fmt.Errorf("failed to extract last intent: %v", err)
-	}
-	newIntent, err := p.getNewIntent()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get new intent: %v", err)
+		return nil, fmt.Errorf("failed to convert patch to typed object: %v", err)
 	}
 
-	element, err := parse.CreateElement(lastIntent, newIntent, current, p.model)
+	// XXX: We don't have a force-flag yet, hence hard-coded to false.
+	outputTyped, managed, err := p.updater.Apply(currentTyped, newTyped, managed, applyManager, false)
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse elements: %v", err)
-	}
-	result, err := element.Merge(strategy.Create(strategy.Options{}))
-	if err != nil {
-		return nil, fmt.Errorf("failed to merge elements: %v", err)
+		return nil, err
 	}
 
-	output, err := p.convertResultToUnversioned(result)
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert merge result: %v", err)
+	output, ok := outputTyped.AsValue().ToUnstructured(true).(map[string]interface{})
+	if !ok {
+		return nil, errors.New("Unable to convert typed unstructured to object unstructured")
 	}
+	newObj := unstructured.Unstructured{Object: output}
 
-	if err := p.saveNewIntent(newIntent, workflowId, output); err != nil {
-		return nil, fmt.Errorf("failed to save last intent: %v", err)
+	if err := apply.EncodeObjectManagedFields(newObj, managed); err != nil {
+		return nil, err
 	}
 
-	// TODO(apelisse): Check for conflicts with other managedFields
-	// and report actionable errors to users.
-
-	return output, nil
+	return &newObj, nil
 }
 
 func (p *applyPatcher) createNewObject() (runtime.Object, error) {
-	original := p.restPatcher.New()
-	objToCreate, gvk, err := p.codec.Decode(p.patchBytes, &p.kind, original)
-	if err != nil {
-		return nil, transformDecodeError(p.typer, err, original, gvk, p.patchBytes)
-	}
-	if gvk.GroupVersion() != p.kind.GroupVersion() {
-		return nil, errors.NewBadRequest(fmt.Sprintf("the API version in the data (%s) does not match the expected API version (%v)", gvk.GroupVersion().String(), p.kind.GroupVersion().String()))
-	}
+	// TODO(kwiesmueller): Need to get the newObject here
+	return p.applyPatchToCurrentObject(nil, p.restPatcher.New())
```

Comment:
That seems dubious, don't you expect the first parameter to be the patch to apply? You're passing nil and empty object here, so that's not going to do much